generator/generator_tests/common.cpp

```diff
@@ -17,11 +21,54 @@ OsmElement MakeOsmElement(uint64_t id, Tags const & tags, OsmElement::EntityType
   return el;
 }
 
-std::string GetFileName()
+std::string GetFileName(std::string filename)
 {
   auto & platform = GetPlatform();
   auto const tmpDir = platform.TmpDir();
   platform.SetWritableDirForTests(tmpDir);
-  return platform.TmpPathForFile();
+  return filename.empty() ? platform.TmpPathForFile() : platform.TmpPathForFile(filename);
```

Comment:
so, ```platform.TmpPathForFile(filename);```?

Reply:
ah, I see the difference now, ok

won't a user be confused which one to use?

Reply:
Why?  

 `std::string TmpPathForFile() `
makes random filename
` std::string TmpPathForFile(std::string const & file)`
makes file by name
I want my wrapper make random file if argument string is empty

Reply:
Platform's interface is ok.
With your interface, ``` GetFileName(s)``` calls different platform's functions depending on the contents of s. This may lead to a confusion.

Reply:
Using generator_tests::GetFileName with empty string will cause a error, what confusion do you expect?

Reply:
Why will it cause an error?
As far as I can see it will call platform's TmpPathForFile with no arguments and return a random file.

(I agree that an error would likely be expected by the caller in this case)